src/color_picker.coffee

```diff
@@ -205,9 +212,16 @@ Ember.Widgets.ColorPicker = Ember.Component.extend
       ])
     ])
 
+  setupCustomColor: Ember.observer(->
+    selectedColor = colorToHex(@get('selectedColor'))
+    return if @get('colorRows').find (row) ->
+      selectedColor in row.invoke 'toLowerCase'
+    @set 'customColor', @get('selectedColor')
+  ).on 'init'
+
   setCustomColor: Ember.observer ->
     if @get('isCustomColorValid') is true
-      @set 'selectedColor', @get 'customColor'
+      @set 'selectedColor', expandHexColor(@get('customColor'))
```

Comment:
Added some stuff here to deal with 3-digit hex codes. Basically, I think `selectedColor` should *always* be a valid, 6-digit code while `customColor` can be whatever the user inputs.